pkg/utils/bpf-iter-ns/bpf-iter-ns.go

```diff
@@ -96,16 +96,22 @@ func ReadOnHostPidNs(iter *link.Iter) ([]byte, error) {
 		return nil, fmt.Errorf("empty /proc/self symlink")
 	}
 
-	tmpPinDir, err := os.MkdirTemp(host.HostRoot+"/sys/fs/bpf", "ig-iter-")
+	// Create a temporary directory in the host bpffs
+	bpfFS := "/sys/fs/bpf"
+	tmpPinDir, err := os.MkdirTemp(host.HostRoot+bpfFS, "ig-iter-")
 	if err != nil {
 		return nil, fmt.Errorf("creating temporary directory in bpffs: %w", err)
 	}
+	defer os.RemoveAll(tmpPinDir)
+
+	// Prepare the pin path from the container and host point of view
 	pinPath := filepath.Join(tmpPinDir, "iter")
+	pinPathHost := filepath.Join(bpfFS, filepath.Base(tmpPinDir), "iter")
```

Comment:
I got a bit confused, I thought `Host` meant it had the `/host` prefix, but it means it's from the host point of view. it doesn't seem a big deal to me.